### DISCUSSION TITLE
What does best_image_timeout do?

### DISCUSSION THREAD

**Discussion by tgiffney** [blakeblackshear/frigate#4446] [Ask A Question]
What does best_image_timeout do?

What are its units (is it in frames or seconds)?

Is it that it controls how often, within an event, the event snapshot (and score used in list of events?) can be replaced if there was a higher scoring detection? Is it like this:
_When an object is detected as a true positive, its image is saved as a snapshot and that snapshot and score appears in the list of events. If the event continues longer than best_image_timeout, once best_image_timeout has elapsed any higher scoring detections will overwrite the snapshot (under clips) and score and reset best_image_timeout?_ 
?

Thanks.

**Comment by NickM-27** [comment]
Pretty close, you can see the logic here: https://github.com/blakeblackshear/frigate/blob/dev/frigate/object_processing.py#L545

Basically, if the score is higher than previously the snapshot / thumbnail will be updated. If the amount of time (in seconds) since the last snapshot is > best_image_timeout one is taken as well.

**Reply by tgiffney** [reply]
Thanks. 
Seems like I had it the wrong way round. I thought `best_image_timeout` was limiting the rate of snapshot/best score updates, when instead it is limiting the outdatedness of snapshots/best scores. 
So after a long event has finished, the saved thumbnail will always be from the last `best_image_timeout` seconds of the recording?  
And otherwise replaced based on the factors in is_better_thumbnail? (meaning that a larger thumbnail could easily replace a higher scoring smaller thumbnail)
And is the value displayed in the web interface related to the "best thumbnail" or instead the maximum score recorded in the whole event?
![frigate](https://user-images.githubusercontent.com/40283149/202881519-0d427544-729e-4214-a11f-63d232833a45.png)
